implicit/als.py

```diff
@@ -107,7 +107,8 @@ def recommend(self, userid, user_items, N=10, filter_items=None, recalculate_use
     def _user_factor(self, userid, user_items, recalculate_user=False):
         if not recalculate_user:
             return self.user_factors[userid]
-        return user_factor(self.item_factors, self.YtY, user_items, userid,
+        return user_factor(self.item_factors, self.YtY,
+                           user_items.tocsr(), userid,
```

Comment:
@benfred Sorry, but the implementation of #35 was not totally correct. It  was missing `.tocsr()` here.

I think nonzeros implementation is not correct if you transpose the matrix:

```
In [6]: counts = csr_matrix([[1, 1, 0, 1, 0, 0],
   ...:                              [0, 1, 1, 1, 0, 0],
   ...:                              [1, 4, 1, 0, 7, 0],
   ...:                              [1, 1, 0, 0, 0, 0],
   ...:                              [9, 0, 4, 1, 0, 1],
   ...:                              [0, 1, 0, 0, 0, 1],
   ...:                              [0, 0, 2, 0, 1, 1]], dtype=np.float64)

In [7]: list(nonzeros((2 * counts.T).tocsr(), 0))
Out[7]: [(0, 2.0), (2, 2.0), (3, 2.0), (4, 18.0)]

In [8]: list(nonzeros((2 * counts.T), 0))
Out[8]: [(0, 2.0), (1, 2.0), (3, 2.0)]
```